src/nupic/types/Serializable.hpp

```diff
@@ -35,12 +35,69 @@
 #include <fstream>
 #include <nupic/os/Directory.hpp>
 #include <nupic/os/Path.hpp>
+#include <nupic/utils/Log.hpp>
 #include <nupic/os/ImportFilesystem.hpp>
 
-#define SERIALIZABLE_VERSION 1
+#define CEREAL_SAVE_FUNCTION_NAME save_ar
+#define CEREAL_LOAD_FUNCTION_NAME load_ar
+#include <cereal/cereal.hpp>
+#include <cereal/archives/binary.hpp>
+#include <cereal/archives/portable_binary.hpp>
+#include <cereal/archives/xml.hpp>
+
+// The RapidJson distribution (Used by Cereal) had a problem with these warnings.
+// It is being fixed in next release.
```

Comment:
nit, pls replace next with the release number, so we can fix it later, 
make a //TODO there

Reply:
> release number, so we can fix it later,

Sure.